python/lsst/afw/math/interpolate.cc

```diff
@@ -22,15 +22,62 @@
 
 #include <pybind11/pybind11.h>
 //#include <pybind11/operators.h>
-//#include <pybind11/stl.h>
+#include <pybind11/stl.h>
+
+#include "numpy/arrayobject.h"
+#include "ndarray/pybind11.h"
+#include "ndarray/converter.h"
+
+#include "lsst/afw/math/interpolate.h"
 
 namespace py = pybind11;
+using namespace pybind11::literals;
 
 using namespace lsst::afw::math;
 
+PYBIND11_DECLARE_HOLDER_TYPE(MyType, std::shared_ptr<MyType>);
+
 PYBIND11_PLUGIN(_interpolate) {
     py::module mod("_interpolate", "Python wrapper for afw _interpolate library");
 
+    if (_import_array() < 0) {
+            PyErr_SetString(PyExc_ImportError, "numpy.core.multiarray failed to import");
+            return nullptr;
+        }
+
+    py::class_<Interpolate, std::shared_ptr<Interpolate>> clsInterpolate(mod, "Interpolate");
+    py::enum_<Interpolate::Style>(clsInterpolate, "Style")
+        .value("UNKNOWN", Interpolate::Style::UNKNOWN)
+        .value("CONSTANT", Interpolate::Style::CONSTANT)
+        .value("LINEAR", Interpolate::Style::LINEAR)
+        .value("NATURAL_SPLINE", Interpolate::Style::NATURAL_SPLINE)
+        .value("CUBIC_SPLINE", Interpolate::Style::CUBIC_SPLINE)
+        .value("CUBIC_SPLINE_PERIODIC", Interpolate::Style::CUBIC_SPLINE_PERIODIC)
+        .value("AKIMA_SPLINE", Interpolate::Style::AKIMA_SPLINE)
+        .value("AKIMA_SPLINE_PERIODIC", Interpolate::Style::AKIMA_SPLINE_PERIODIC)
+        .value("NUM_STYLES", Interpolate::Style::NUM_STYLES)
+        .export_values();
+
+    clsInterpolate.def("interpolate", [](Interpolate &t, double const x) -> double {
```

Comment:
Return type deduction will work fine here. Only put it in when needed (e.g. multiple branches or different type, etc.).